wrappers/kn/src/nativeMain/kotlin/zxingcpp/ImageView.kt

```diff
@@ -0,0 +1,52 @@
+package zxingcpp
+
+import cnames.structs.ZXing_ImageView
+import kotlinx.cinterop.*
+import zxingcpp.cinterop.*
+import kotlin.experimental.ExperimentalNativeApi
+import kotlin.native.ref.createCleaner
+
+@OptIn(ExperimentalForeignApi::class)
+class ImageView(
+	val data: ByteArray,
+	val width: Int,
+	val height: Int,
+	val format: ImageFormat,
+	val rowStride: Int = 0,
+	val pixStride: Int = 0,
+) {
+	private val pinnedData = data.pin()
+	val cValue: CPointer<ZXing_ImageView>? =
+		ZXing_ImageView_new(
+			pinnedData.addressOf(0).reinterpret(),
+			width,
+			height,
+			format.rawValue,
+			rowStride,
+			pixStride
+		)
+
+	@Suppress("unused")
+	@OptIn(ExperimentalNativeApi::class)
+	private val cValueCleaner = createCleaner(cValue) { ZXing_ImageView_delete(it) }
+
+	@Suppress("unused")
+	@OptIn(ExperimentalNativeApi::class)
+	private val pinnedDataCleaner = createCleaner(pinnedData) { it.unpin() }
```

Comment:
I believe one 'cleaner' variable that calls both cleanup functions on destruction is sufficient.

Reply:
As you can see, the cleaner holds reference to the object have to be cleaned.
At the stage of cleaning, the reference in object itself is not reliable.
And pass `this` to createCleaner will cause mem leak as described in document of `cleaner`

Reply:
You are right. Sorry...